examples/server/server.cpp

```diff
@@ -1162,6 +1164,16 @@ struct server_metrics {
     }
 };
 
+struct termination_signal {
+    int number;
+};
+
+struct standby_timeout {};
+
+using shutdown_reason = std::variant<termination_signal, standby_timeout>;
```

Comment:
Just an enum is enough, and will be more aligned with your coding style. Please read the CONTRIBUTING.md

> Avoid fancy-looking modern STL constructs, use basic for loops, avoid templates, keep it simple